pallets/creditcoin/src/migrations/mod.rs

```diff
@@ -9,55 +10,33 @@ mod v5;
 mod v6;
 mod v7;
 
-#[cfg(feature = "try-runtime")]
-pub(crate) fn pre_upgrade<T: Config>(version: StorageVersion) -> Result<(), &'static str> {
-	// potentially do some checks on the state before the migration
-	if version < 6 {
-		v6::pre_upgrade::<T>()?;
-	}
-	Ok(())
-}
-
-#[cfg(feature = "try-runtime")]
-pub(crate) fn post_upgrade<T: Config>(version: StorageVersion) -> Result<(), &'static str> {
-	// do some post-migration checks
-	if version < 6 {
-		v6::post_upgrade::<T>()?;
-	}
-	Ok(())
-}
-
 pub(crate) fn migrate<T: Config>() -> Weight {
 	let version = StorageVersion::get::<Pallet<T>>();
 	let mut weight: Weight = Weight::zero();
 
-	if version < 1 {
-		weight.saturating_accrue(v1::migrate::<T>());
-		StorageVersion::new(1).put::<Pallet<T>>();
-	}
-	if version < 2 {
-		weight.saturating_accrue(v2::migrate::<T>());
-		StorageVersion::new(2).put::<Pallet<T>>();
-	}
-	if version < 3 {
-		weight.saturating_accrue(v3::migrate::<T>());
-		StorageVersion::new(3).put::<Pallet<T>>();
-	}
-	if version < 4 {
-		weight.saturating_accrue(v4::migrate::<T>());
-		StorageVersion::new(4).put::<Pallet<T>>();
-	}
-	if version < 5 {
-		weight.saturating_accrue(v5::migrate::<T>());
-		StorageVersion::new(5).put::<Pallet<T>>();
-	}
-	if version < 6 {
-		weight.saturating_accrue(v6::migrate::<T>());
-		StorageVersion::new(6).put::<Pallet<T>>();
-	}
-	if version < 7 {
-		weight.saturating_accrue(v7::migrate::<T>());
-		StorageVersion::new(7).put::<Pallet<T>>();
+	let callbacks = HashMap::from([
+		(1, (v1::pre_upgrade::<T>, v1::migrate::<T>, v1::post_upgrade::<T>)),
+		(2, (v2::pre_upgrade::<T>, v2::migrate::<T>, v2::post_upgrade::<T>)),
+		(3, (v3::pre_upgrade::<T>, v3::migrate::<T>, v3::post_upgrade::<T>)),
+		(4, (v4::pre_upgrade::<T>, v4::migrate::<T>, v4::post_upgrade::<T>)),
+		(5, (v5::pre_upgrade::<T>, v5::migrate::<T>, v5::post_upgrade::<T>)),
+		(6, (v6::pre_upgrade::<T>, v6::migrate::<T>, v6::post_upgrade::<T>)),
+		(7, (v7::pre_upgrade::<T>, v7::migrate::<T>, v7::post_upgrade::<T>)),
+	]);
```

Comment:
I tried following a suggestion from Carlos but can't seem to figure out the compiler errors.

Reply:
try [this](https://github.com/gluwa/creditcoin/tree/feature/pre-post-migration-checks) impl.